src/__tests__/utils/testForm.tsx

```diff
@@ -3,8 +3,18 @@ import { Control, useController } from "react-hook-form";
 import { z } from "zod";
 import { createUniqueFieldSchema } from "../../createFieldSchema";
 import { createTsForm } from "../../createSchemaForm";
+import { useTsController } from "../../FieldContext";
 
 export const textFieldTestId = "text-field";
+export const defaultTextInputTestId = "text-input";
+export const defaultBooleanInputTestId = "boolean-input";
+export const defaultNumberInputTestId = "number-input";
+export const errorMessageTestId = "error-message";
+
+export function ErrorMessage() {
+  const { error } = useTsController();
+  return <div data-testid={errorMessageTestId}>{error?.errorMessage}</div>;
```

Comment:
this provides a clean way to add error message to any test component which will let me check for no errors consistently. one thing that's weird is that validation via react-hook-form doesn't seem to rerender this hook so i was forced to rerender in tests. it doesn't seem to be a problem IRL but worth mentioning

Reply:
would it make sense to early return null if `error` is not present?

Reply:
good call, way easier / better